giotto/graphs/create_clique_complex.py

```diff
@@ -0,0 +1,387 @@
+
+
+class CreateCliqueComplex:
+    """Pre-processing step from unstructured data to graph and complex.
+
+    Useful in graph analysis to convert general point clouds or distance
+    matrices (numpy objects) into graphs (networkx objects) and clique
+    complex (described as a dictionary).
+
+    Parameters
+    ----------
+    graph : nx.graph, optional, default: ``None``
+        Graph of which computing the clique complex.
+
+    data : ndarray, shape (n_points, n_dimensions) or \
+        (n_points, n_points)
+        If ``data_type == 'cloud'``, the input should be an
+        ndarray whose each entry along axis 0 is a vector with
+        the features of the relative point. Otherwise when
+        'matrix' is passed as data_type the object should be a symmetric
+        square matrix with distances between points.
+
+    alpha : float
+        Real value to be used as a threshold while constructing
+        the alpha-neighboring graph. For a point 'x' there will be an edge
+        to each point 'y' if dist(x,y) < alpha, where here the
+        distance is the euclidean.
+
+    data_type : string, optional, default: ``'graph'``
+        Type of raw data to be preprocessed. If set to 'graph'
+        the Input 'graph' has to be a netowrkx graph object.
+        If set to 'cloud' the data is to be interpreted as a collection
+        of points where row index represents sample ID and columns the features
+        If set to 'matrix' the data is to be interpreted as a
+        distance matrix (symmetric) collecting element-wise distances
+        between the elements.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+    >>> import networkx as nx
+    >>> X = np.random.random((20,2))
+    >>> plt.scatter(X[:,0], X[:,1])
+    >>> cc = CreateCliqueComplex(X, 0.5)
+    >>> cd = cc.create_complex_from_graph()
+    >>> nx.draw(cc.get_graph())
+
+    """
+
+    def __init__(self, graph=None, data=None, alpha=None, data_type='graph'):
+        self.data = data
+        self.alpha = alpha
+        self.data_type = data_type
+        self.complex_dict = dict()
+
+        if data_type == 'graph':
+            # check type graph
+            if not isinstance(graph, nx.Graph):
+                raise ValueError("The parameter graph should be "
+                                "a networkx Graph object")
+            self.graph = graph
+            self.adjacent_matrix = nx.adjacency_matrix(self.graph)
+        else:
+            check_array(data)
+            if self.alpha is None:
+                raise ValueError("If 'data_type' is not 'graph' the parameter"
+                                "alpha must be a float number.")
+            self.graph = self._create_graph()
+            self.adjacent_matrix = csr_matrix(self.adjacent_matrix)
+
+    def create_complex_from_graph(self):
+        """
+        Functions that computes the Cliques Complex starting from
+        the related networkx graph object.
```

Comment:
Suggestion: "Function that computes the Clique complex from the associated networkx Graph object"